text/0103-introduce-core-index-commitment.md

```diff
@@ -0,0 +1,313 @@
+	/// Thus it cannot be derived entirely from the relay-parent.
+	persisted_validation_data_hash: Hash,
+	/// The blake2-256 hash of the PoV.
+	pov_hash: Hash,
+	/// The root of a block's erasure encoding Merkle tree.
+	erasure_root: Hash,
+	/// Reserved bytes.
+	reserved2: [u8; 64],
+	/// Hash of the para header that is being generated by this candidate.
+	para_head: Hash,
+	/// The blake2-256 hash of the validation code bytes.
+	validation_code_hash: ValidationCodeHash,
+}
+```
+
+In future format versions, parts of the `reserved1` and `reserved2` bytes can be used to include 
+additional information in the descriptor.
+
+### Parachain block validation
+
+If the candidate descriptor is version 1, there are no changes.
+
+Backers must check the validity of `core_index` and `session_index` fields. 
+A candidate must not be backed if any of the following are true:
+- the `core_index` in the descriptor does not match the core the backer is assigned to
+- the `session_index` is not equal to the session of the `relay_parent` in the descriptor
+- the `core_index` in the descriptor does not match the one determined by the 
+  `UMPSignal::SelectCore` message
+
+
+### On-chain backing
+
+If the candidate descriptor is version 1, there are no changes.
+
+For version 2 descriptors the runtime will determine the `core_index` similarly to backers but 
+will always ignore the committed claim queue offset and use `0`, as it only cares 
+about what candidates can be backed at current block. 
+
+As the chain advances the claims also advance to the top of the queue. The runtime will only back  
+a candidate if the claimed core selected by it's claim queue offset has reached the top of the queue 
+at the current relay chain block: `current_block_num - relay_parent_num - 1 == claim_queue_offset`.
+
+The impact of this change is that candidates built into the future (`claim queue offset > 0`) 
+can no longer be backed earlier even if the core is free and the core is assigned to the 
+parachain. 
+
+Introducing this additional check is required to ensure the runtime computes the core index
+determinstically. For example, some collator has missed his slot and the core is now used 
+to back a candidate with a higher claim queue offset. The number of assigned cores can
+be different at these two queue offsets and the committed core indices would be different. 
+
+
+### Backwards compatibility
+
+There are two flavors of candidate receipts that are used in network protocols, runtime and node 
+implementation:
+- `CommittedCandidateReceipt` which includes the `CanidateDescriptor` and the `CandidateCommitments` 
+- `CandidateReceipt` which includes the `CanidateDescriptor` and just a hash of the commitments
+
+We want to support both the old and new versions in the runtime and node. The implementation must 
+be able to detect the version of a given candidate receipt.
+
+The version of the descriptor is detected by checking if the `version` field is `0` and the 
```

Comment:
Uh ok. If we really want to do this, then disregard my comment about the version field above.

Reply:
Yeah, it is not reserved, but actually used already.